torchmetrics/functional/text/wer.py

```diff
@@ -12,48 +12,103 @@
+
+    Returns:
+        (Tensor) Number of edit operations to get from the reference to the prediction, summed over all samples
+        (Tensor) Number of words over all references
+    """
+    if isinstance(predictions, str):
+        predictions = [predictions]
+    if isinstance(references, str):
+        references = [references]
+    errors = tensor(0, dtype=torch.float)
+    total = tensor(0, dtype=torch.float)
+    for prediction, reference in zip(predictions, references):
+        prediction_tokens = prediction.split()
+        reference_tokens = reference.split()
+        errors += _edit_distance(prediction_tokens, reference_tokens)
+        total += len(reference_tokens)
+    return errors, total
+
+
+def _wer_compute(errors: Tensor, total: Tensor) -> Tensor:
+    """Compute the word error rate.
+
+    Args:
+        errors: Number of edit operations to get from the reference to the prediction, summed over all samples
+        total: Number of words over all references
+
+    Returns:
+        (Tensor) Word error rate
+    """
+    return errors / total
+
+
+def wer(
     predictions: Union[str, List[str]],
+    references: Union[str, List[str]],
     concatenate_texts: bool = False,
-) -> float:
-    """Word error rate (WER_) is a common metric of the performance of an automatic speech recognition system. This
+) -> Tensor:
+    """Word error rate (WER) is a common metric of the performance of an automatic speech recognition system. This
     value indicates the percentage of words that were incorrectly predicted. The lower the value, the better the
     performance of the ASR system with a WER of 0 being a perfect score.
 
     Args:
-        references: List of references for each speech input.
-        predictions: List of transcriptions to score.
-        concatenate_texts: Whether to concatenate all input texts or compute WER iteratively.
+        predictions: Transcription(s) to score as a string or list of strings
+        references: Reference(s) for each speech input as a string or list of strings
+        concatenate_texts: Whether to concatenate all input texts or compute WER iteratively
+            This argument is deprecated in v0.6 and it will be removed in v0.7.
 
     Returns:
-        (float): the word error rate, or if ``return_measures`` is True, we include the incorrect and total.
+        (Tensor) Word error rate
 
     Examples:
         >>> predictions = ["this is the prediction", "there is an other sample"]
         >>> references = ["this is the reference", "there is another one"]
         >>> wer(predictions=predictions, references=references)
-        0.5
+        tensor(0.5000)
     """
-    if not _JIWER_AVAILABLE:
-        raise ModuleNotFoundError(
-            "wer metric requires that jiwer is installed."
-            " Either install as `pip install torchmetrics[text]` or `pip install jiwer`"
-        )
     if concatenate_texts:
```

Comment:
this will make sure that user that are setting this flag to `False` also will get the warning message
```suggestion
    if concatenate_texts is not None:
```